packages/babel-helper-evaluate-path/src/index.js

```diff
@@ -80,6 +80,10 @@ function evaluateIdentifier(path) {
     }
   }
 
+  if (binding.kind === 'const') {
+    return path.evaluate();
```

Comment:
this will skip the TDZ check below for `const`

Reply:
do you have an code example where this would be a problem ? The solution would be to add `binding.kind !== 'const' &&` in `if (shouldDeoptBasedOnScope(binding, path)) {` ?

Reply:
(sorry about the outdated I force-pushed to fix the tests...)